src/libs/antares/study/parts/thermal/cluster_list.cpp

```diff
@@ -719,9 +719,9 @@ void ThermalClusterList::ensureDataPrepro()
     auto end = cluster.end();
     for (auto it = cluster.begin(); it != end; ++it)
     {
-        auto& c = *(it->second);
-        if (not c.prepro)
-            c.prepro = new PreproThermal();
+        auto c = it->second;
+        if (not c->prepro)
+            c->prepro = new PreproThermal(c.get());
```

Comment:
Storing the underlying address of a smart ptr is **extremely dangerous**. The managed object could be destroyed, leading to a segfault. I strongly suggest that we keep a `std::shared_ptr<const ThermalCluster>` here.

Reply:
Done